functions/java-jre.bash

```diff
@@ -100,95 +99,117 @@ java_zulu_8_tar(){
-  # Fetch an JSON array of download candidates from azul and filter them
-  # shellcheck disable=SC2006
-  downloadlink=`curl 'https://www.azul.com/wp-admin/admin-ajax.php' -s \
-        -H 'Accept: */*' \
-        -H 'X-Requested-With: XMLHttpRequest' \
-        -H 'Content-Type: multipart/form-data; boundary=---------------------------17447165291708986765346780730' \
-        -H 'Connection: keep-alive' \
-        -H 'Cache-Control: max-age=0' -H 'TE: Trailers' \
-        --data-binary $'-----------------------------17447165291708986765346780730\r\nContent-Disposition: form-data; name="action"\r\n\r\nbundles_filter_query\r\n-----------------------------17447165291708986765346780730\r\nContent-Disposition: form-data; name="action"\r\n\r\nsearch_bundles\r\n-----------------------------17447165291708986765346780730--\r\n' \
-        | jq -r "$filter"`
+
   if [ -z "$downloadlink" ]; then return 1; fi
-    # shellcheck disable=SC2046,SC2005,SC2006
-    echo `echo "$downloadlink" | head -n1`
+  echo "$downloadlink"
   return 0
 }
 
-## Check whatever a newer version Zulu version is available.
-## Returns 0 / true if new version exist
-## TODO: Rewrite using download API when it leaves alpha state. https://www.azul.com/downloads/zulu-community/api/
+## Check if a newer version of Java Zulu 8 is available.
+## Returns 0 / true if new version exists
+## Valid arguments: 64-bit, 32-bit
+##
+##    java_zulu_tar_update_available(String arch)
+##
 java_zulu_tar_update_available(){
   if [ ! -x "$(command -v java)" ]; then return 0; fi
-
-  local availableVersion # format: "8u222-b10"
+  local availableVersion
   local javaVersion
   local filter
+  local link
   if [ ! -x "$(command -v jq)" ]; then
-    apt-get install -y jq
+    cond_redirect apt-get install --yes jq
+  fi
+
+  filter='[.zulu_version[] | tostring] | join(".")'
+  link="https://api.azul.com/zulu/download/community/v1.0/bundles/latest/?jdk_version=8&ext=tar.gz&os=linux"
+  javaVersion=$(java -version 2>&1 | grep -m 1 -o "[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}")
+
+  if [ "$1" == "32-bit" ]; then
+    if is_arm; then
+      availableVersion=$(curl -s -H "Accept: application/json" "${link}&arch=arm&hw_bitness=32" | jq -r "$filter")
+    else
+      availableVersion=$(curl -s -H "Accept: application/json" "${link}&arch=x86&hw_bitness=32" | jq -r "$filter")
+    fi
+
+  elif [ "$1" == "64-bit" ]; then
+    if is_arm; then
+      availableVersion=$(curl -s -H "Accept: application/json" "${link}&arch=arm&hw_bitness=64" | jq -r "$filter")
+    else
+      availableVersion=$(curl -s -H "Accept: application/json" "${link}&arch=x86&hw_bitness=64" | jq -r "$filter")
+    fi
+
+  else
+    echo -n "[DEBUG] Invalid argument to function java_zulu_tar_update_available()"
```

Comment:
do not output [DEBUG] but "[$timestamp] Installing XXX ...." at the beginning and "OK" or "FAILED" at the end to be consistent with rest of code.